src/frontend/js/shoppingcart/reducer.js

```diff
@@ -14,7 +14,7 @@ export const computeShoppingCartTotal = products =>
     products.reduce((total, p) => total + (p.price * p.quantity), 0);
 
 export default (state = initialState, { type, payload }) => {
-    const products = state.products;
+    const products = [].concat(state.products);
```

Comment:
What it is for ? Isolation ?
I don't know if `[ ...state.products ]` is preferable, but not very important!